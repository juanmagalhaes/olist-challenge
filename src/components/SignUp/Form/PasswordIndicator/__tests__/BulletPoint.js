import React from 'react';
import {
  Bullet,
  BulletPoint,
  BulletPoints,
} from '../BulletPoint';

describe('Bullet', () => {
  const shallowWrapper = props => shallow(<Bullet {...props} />);

  test('renders', () => {
    const wrapper = shallowWrapper({
      isEmpty: true,
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('renders with success style', () => {
    const wrapper = shallowWrapper({
      isEmpty: false,
      isValid: true,
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('renders with danger style', () => {
    const wrapper = shallowWrapper({
      isEmpty: false,
    });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('BulletPoint', () => {
  const shallowWrapper = props => shallow(<BulletPoint {...props} />);

  test('renders', () => {
    const wrapper = shallowWrapper({
      name: 'foo',
      isEmpty: true,
      validation: {
        foo: {
          isValid: true,
          msg: 'bar',
        },
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

});

describe('BulletPoints', () => {
  const shallowWrapper = props => shallow(<BulletPoints {...props} />);

  test('renders', () => {
    const wrapper = shallowWrapper({
      isEmpty: true,
      passwordStrength: {},
    });
    expect(wrapper).toMatchSnapshot();
  });

});

