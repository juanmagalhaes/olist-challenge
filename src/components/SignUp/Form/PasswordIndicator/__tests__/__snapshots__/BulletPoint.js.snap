// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bullet renders 1`] = `
<div
  className="bullet"
/>
`;

exports[`Bullet renders with danger style 1`] = `
<div
  className="bullet bulletDanger"
/>
`;

exports[`Bullet renders with success style 1`] = `
<div
  className="bullet bulletSuccess"
/>
`;

exports[`BulletPoint renders 1`] = `
<div
  className="bulletPoint"
>
  <Bullet
    isEmpty={true}
    isValid={true}
  />
  <span
    className="text"
  >
    bar
  </span>
</div>
`;

exports[`BulletPoints renders 1`] = `
<div
  className="bullets"
>
  <BulletPoint
    isEmpty={true}
    key="0"
    name="minLength"
    validation={Object {}}
  />
  <BulletPoint
    isEmpty={true}
    key="1"
    name="containsUppercase"
    validation={Object {}}
  />
  <BulletPoint
    isEmpty={true}
    key="2"
    name="containsNumber"
    validation={Object {}}
  />
</div>
`;
